From 00b81e9e66db7812ef83d0641e72d85bd02aa0a8 Mon Sep 17 00:00:00 2001
From: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
Date: Wed, 31 Jul 2019 11:48:29 +0530
Subject: [PATCH] Fix for Pairing re-initiated after Bluetooth Reset

Issue:  Pairing is re-initiated with previously paired
devices after Successful Network Reset

Reason:
- Once we pair & connect reference device to IVI, and
  perform BT Factory reset which disables bluetooth
  service and clear configurations.After completion of
  reset,Bluetooth Service comes up and sends connect
  request to reference phone,but ideally it should not
  have any info of previously paired device.
- In BT Adapter, at the time of reset it only disables
  BT Service and clear bt_config file, but nowhere it
  removes bond from paired devices, due to which IVI
  starts autoconnect for previously paired device.

Fix:
- Remove Bonded devices at the time of BT Factory reset
  in BT Adapter. So that it will remove bond and then
  disable Bluetooth Service.

Test:
- Connected Pixel2 with IVI,triggered reset from Car
  Settings app.As expected, after reset completion there
  was no reconnect request sent for previously paired
  devices.

Change-Id: I6163756c1f6d9339110a502fffd0b7f6c6ae23dd
Tracked-On:
Signed-off-by: Gaganpreet kaur <gaganpreetx.kaur@intel.com>
---
 src/com/android/bluetooth/btservice/AdapterService.java | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/com/android/bluetooth/btservice/AdapterService.java b/src/com/android/bluetooth/btservice/AdapterService.java
index 508eacf..515469b 100644
--- a/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/src/com/android/bluetooth/btservice/AdapterService.java
@@ -1559,6 +1559,14 @@ public class AdapterService extends Service {
             if (service == null) {
                 return false;
             }
+
+            BluetoothDevice[] bondedDeviceList = getBondedDevices();
+            if (bondedDeviceList != null) {
+                for (BluetoothDevice device : bondedDeviceList) {
+                    service.removeBond(device);
+                }
+            }
+
             service.disable();
             return service.factoryReset();
 
-- 
2.7.4

